File: Java/017.java
public class Problem017 {

    private static final String[] digitsToWord = {
        "", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"
    };

    private static final String[] teensToWord = {
        "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"
    };

    private static final String[] tensToWord = {
        "", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"
    };

    public static String numToWord(int n) {
        if (n == 1000) {
            return "onethousand";
        }

        StringBuilder res = new StringBuilder();

        int hundreds = n / 100;
        if (hundreds > 0) {
            res.append(digitsToWord[hundreds]).append("hundred");
            if (n % 100 != 0) {
                res.append("and");
            }
        }

        int remainder = n % 100;
        if (remainder >= 10 && remainder < 20) {
            res.append(teensToWord[remainder - 10]);
        } else {
            int tens = remainder / 10;
            int ones = remainder % 10;
            if (tens > 0) {
                res.append(tensToWord[tens]);
            }
            if (ones > 0) {
                res.append(digitsToWord[ones]);
            }
        }

        return res.toString();
    }

    public static void main(String[] args) {
        int res = 0;
        for (int n = 1; n <= 1000; n++) {
            res += numToWord(n).length();
        }
        System.out.println(res);
    }
}

File: Rust/017.rs
const DIGITS_TO_WORD: [&str; 10] = [
    "", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine",
];
const TEENS_TO_WORD: [&str; 10] = [
    "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen",
    "eighteen", "nineteen",
];
const TENS_TO_WORD: [&str; 10] = [
    "", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety",
];

fn num_to_word(n: u32) -> String {
    if n == 1000 {
        return "onethousand".to_string();
    }

    let mut res = String::new();
    let hundreds = n / 100;
    let mut remainder = n % 100;

    if hundreds > 0 {
        res.push_str(DIGITS_TO_WORD[hundreds as usize]);
        res.push_str("hundred");
        if remainder > 0 {
            res.push_str("and");
        }
    }

    if remainder >= 10 && remainder < 20 {
        res.push_str(TEENS_TO_WORD[(remainder - 10) as usize]);
    } else {
        let tens = remainder / 10;
        let ones = remainder % 10;
        if tens > 0 {
            res.push_str(TENS_TO_WORD[tens as usize]);
        }
        if ones > 0 {
            res.push_str(DIGITS_TO_WORD[ones as usize]);
        }
    }
    res
}

fn main() {
    let mut total_letters = 0;
    for n in 1..=1000 {
        let word = num_to_word(n);
        total_letters += word.len();
    }
    println!("{}", total_letters);
}
